components/builder-api/src/server/handlers.rs

```diff
@@ -181,6 +181,44 @@ pub fn job_group_promote(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+pub fn job_group_cancel(req: &mut Request) -> IronResult<Response> {
+    let group_id = match get_param(req, "id") {
+        Some(id) => {
+            match id.parse::<u64>() {
+                Ok(g) => g,
+                Err(e) => {
+                    debug!("Error finding group. e = {:?}", e);
+                    return Ok(Response::with(status::BadRequest));
+                }
+            }
+        }
+        None => return Ok(Response::with(status::BadRequest)),
+    };
```

Comment:
Not something that needs to be done in this PR, but I'd _love_ to find a way to extract this pattern into some kind of abstraction.  I feel like we have a ton of "find this param and return a 400 if it's not present", or "find this param and parse it into an int and return a 400 if it's not present" sprinkled throughout the depot/api components.  Maybe something we can make an issue for and tackle in the future.

Reply:
yeah totally - this aspect of the handlers is ripe for some further cleanup.